driver-benchmarks/src/main/com/mongodb/benchmark/benchmarks/GridFSImportBenchmark.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.benchmark.benchmarks;
+
+import com.mongodb.benchmark.framework.BenchmarkResult;
+import com.mongodb.benchmark.framework.BenchmarkRunner;
+import com.mongodb.benchmark.framework.TextBasedBenchmarkResultWriter;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class GridFSImportBenchmark extends AbstractMongoBenchmark {
+    private MongoDatabase database;
+    private GridFSBucket bucket;
+
+    private ExecutorService fileService;
+
+    @Override
+    public String getName() {
+        return "GridFS multi-file upload";
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        database = client.getDatabase("perftest");
+        bucket = GridFSBuckets.create(database);
```

Comment:
Do we have to use the default chunksize? - I imagine it will have a large impact on the results for GridFS.

Reply:
I don't think we do.  What would you recommend?

Reply:
Done